awa-python/python/awa/client.py

```
@@ -319,6 +366,132 @@ async def list_jobs(
+    # --- Durable batch operations -----------------------------------------
+    #
+    # Batch operations are maintenance-led admin mutations. They snapshot an
+    # eligible job set, apply changes in small chunks, and keep progress /
+    # cancellation state in `awa.batch_operations`.
+
+    async def preview_batch_operation(
+        self,
+        op_kind: BatchOperationKind,
+        *,
+        spec: dict[str, Any],
+        filter: BatchOperationFilter | None = None,
```

Ruff (0.15.15)

[error] Function argument `filter` is shadowing a Python builtin

(A002)